### ISSUE TITLE
Regression in 0.11.3 -- NRE from ILProcessor.InsertBefore()

### ISSUE THREAD

**Issue by SteveGilham** [jbevain/cecil#697]
```
System.NullReferenceException
  HResult=0x80004003
  Message=Object reference not set to an instance of an object.
  Source=Mono.Cecil
  StackTrace:
   at Mono.Cecil.Cil.InstructionCollection.ResolveInstructionOffset(InstructionOffset inputOffset, InstructionOffsetCache& cache) in C:\sources\cecil\Mono.Cecil.Cil\MethodBody.cs:line 412
   at Mono.Cecil.Cil.InstructionCollection.UpdateLocalScope(ScopeDebugInformation scope, Instruction removedInstruction, Instruction existingInstruction, InstructionOffsetCache& cache) in C:\sources\cecil\Mono.Cecil.Cil\MethodBody.cs:line 345
   at Mono.Cecil.Cil.InstructionCollection.UpdateLocalScope(ScopeDebugInformation scope, Instruction removedInstruction, Instruction existingInstruction, InstructionOffsetCache& cache) in C:\sources\cecil\Mono.Cecil.Cil\MethodBody.cs:line 356
   at Mono.Cecil.Cil.InstructionCollection.UpdateLocalScopes(Instruction removedInstruction, Instruction existingInstruction) in C:\sources\cecil\Mono.Cecil.Cil\MethodBody.cs:line 334
   at Mono.Collections.Generic.Collection`1.Insert(Int32 index, T item) in C:\sources\cecil\Mono.Collections.Generic\Collection.cs:line 149
```

Simple repro attached

[Cecil.11.3.issue.zip](https://github.com/jbevain/cecil/files/5377742/Cecil.11.3.issue.zip)

**Comment by SteveGilham** [comment]
The null reference is the 7th of the 6 instructions in the method body; the initial access to the instructions through the `Collection<Instruction>` type with size less than capacity decays to a direct access to the internal `Instruction[]` which has length equal to the capacity.

FWIW, the assembly was built with an old Mono compiler, so advertises a code size of 14, hence the capacity of 7, for 6 instructions.

**Comment by jbevain** [comment]
Fixed in #698.
